torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -127,22 +141,24 @@ struct GraphFuser {
       return false;
     }
   }
-  bool allFloatIO(Node * node) {
-    for(auto & o : node->outputs()) {
-      if(!hasFloatType(o)) {
-        return false;
-      }
-    }
-    for(auto & o : node->inputs()) {
+  bool allFloatList(at::ArrayRef<Value*> list){
+    for (auto & o: list){
       if(!hasFloatType(o)) {
         return false;
       }
     }
     return true;
   }
+  bool allFloatIO(Node * node) {
+    if (!allFloatList(node->outputs())) return false;
+    if (!allFloatList(node->inputs())) return false;
+    return true;
+  }
   bool isFusable(Node * node) {
     if (node->owningBlock() != block) return false;
     if (node->kind() == prim::FusionGroup) return true;
+    if (ignoreOutputType(node)) return isSimpleMap(node) && allFloatList(node->inputs());
+    if (node->kind() == aten::type_as) return isSimpleMap(node) && allFloatList(node->outputs());
     return isSimpleMap(node) && allFloatIO(node);
```

Comment:
Can we simply do `if (!isSimpleMap(node)) return false;` above, and do the type checks below?

Reply:
Then, once you simplify `isSimpleMap(node)` we can have `typeCheck(node)` which looks like
```cpp
switch (node->kind()) {
  case aten::le:
  ...
  case aten::eq:
    return allFloat(node->inputs());
  case aten::type_as:
    return allFloat(node->outputs());
  default:
    return allFloat(node->inputs()) && allFloat(node->outputs());
}
```